src/compiler/checker.ts

```diff
@@ -30826,11 +30823,31 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function getJsxElementTypeTypeAt(location: Node): Type | undefined {
-        const type = getJsxType(JsxNames.ElementType, location);
-        if (isErrorType(type)) return undefined;
+        const ns = getJsxNamespaceAt(location);
+        if (!ns) return undefined;
+        const sym = getJsxElementTypeSymbol(ns);
+        if (!sym) return undefined;
+        const type = instantiateAliasOrInterfaceWithDefaults(sym, isInJSFile(location));
+        if (!type || isErrorType(type)) return undefined;
         return type;
     }
 
+    function instantiateAliasOrInterfaceWithDefaults(managedSym: Symbol, inJs: boolean, ...typeArguments: Type[]) {
+        const declaredManagedType = getDeclaredTypeOfSymbol(managedSym); // fetches interface type, or initializes symbol links type parmaeters
+        if (managedSym.flags & SymbolFlags.TypeAlias) {
+            const params = getSymbolLinks(managedSym).typeParameters;
+            if (length(params) >= typeArguments.length) {

```

Comment:
a type alias with all its type arguments provided will return `undefined` from this function. Why not return the type alias itself? Does getJsxElementTypeTypeAt only return a type if the JSX.ElementType is a type alias with defaulted parameters?

Looking at getJsxManagedAttributesFromLocatedAttributes, all these questions really originate there, because this code is abstracted from it.

Reply:
> a type alias with all its type arguments provided will return undefined from this function.

...No? A type alias with an insufficient number of arguments to handle the `typeArguments` passed in (which is zero for `ElementType`, so any amount will do, but 2 for `LibraryManagedAttributes`) will return `undefined.

Reply:
It took a lot of staring, but I figured out that: I misread `>=` as `>`.